GrasshopperAsyncComponent/GH_AsyncComponent.cs

```diff
@@ -170,31 +190,30 @@ protected override void SolveInstance(IGH_DataAccess da)
 
             // Add cancellation source to our bag
             var tokenSource = new CancellationTokenSource();
-            CancellationSources.Add(tokenSource);
 
             var currentWorker = BaseWorker.Duplicate($"Worker-{da.Iteration}", tokenSource.Token);
-            if (currentWorker == null)
-            {
-                AddRuntimeMessage(GH_RuntimeMessageLevel.Error, "Could not get a worker instance.");
-                return;
-            }
 
             // Let the worker collect data.
             currentWorker.GetData(da, Params);
 
-            var currentRun =
-                TaskCreationOptions != null
-                    ? new Task(
-                        () => currentWorker.DoWork(_reportProgress, _done),
-                        tokenSource.Token,
-                        (TaskCreationOptions)TaskCreationOptions
-                    )
-                    : new Task(() => currentWorker.DoWork(_reportProgress, _done), tokenSource.Token);
+            var currentRun = new Task<Task>(
```

Comment:
Task.Factory.StartNew()

Reply:
No try/catch or other error handling for this task?  Unobserved task exceptions 😬

Reply:
It's important that the tasks don't start within this SolveInstance.
They get started all together in PostSolveInstance

That way, it it guaranteed that a task never calls `Done` while we are still spinning up workers. That would break the components state tracking.

Reply:
Perhaps there's alternatives with storing `Func<Task>`, but for this PR I wanted to keep minimal changes